src/query/sql/src/planner/binder/binder.rs

```diff
@@ -674,6 +681,150 @@ impl<'a> Binder {
         Ok(finder.scalars().is_empty())
     }
 
+    pub(crate) fn check_sexpr_top(&self, s_expr: &SExpr, top_check: CheckType) -> Result<bool> {
```

Comment:
Better to have comments to explain what's the functions did, and note the user may attentation if you want.